cmd/alias-set.go

```diff
@@ -184,45 +194,81 @@ func probeS3Signature(ctx context.Context, accessKey, secretKey, url string) (st
 		Debug:     globalDebug,
 	}
 
-	probeSignatureType := func(stype string) (string, *probe.Error) {
-		s3Config.Signature = stype
-		s3Client, err := S3New(s3Config)
-		if err != nil {
-			return "", err
+	var (
+		s3Client Client
+		pErr     *probe.Error
+	)
+	APIs := []string{"s3v4", "s3v2"}
+	for len(APIs) > 0 {
+		s3Config.Signature = APIs[0]
+		s3Client, pErr = newFactory()(s3Config)
+		if pErr != nil {
+			return "", pErr
+		}
+		if _, pErr = s3Client.Stat(ctx, StatOptions{}); pErr == nil {
+			return s3Config.Signature, nil
 		}
 
-		if _, err := s3Client.Stat(ctx, StatOptions{}); err != nil {
-			e := err.ToGoError()
-			if _, ok := e.(BucketDoesNotExist); ok {
-				// Bucket doesn't exist, means signature probing worked successfully.
-				return stype, nil
+		if strings.Contains(pErr.ToGoError().Error(), "x509: certificate signed by unknown authority") {
+			// We don't need to fetch the peer certificate
+			// if STDOUT is not a terminal and the user cannot
+			// confirm the public key.
+			if !term.IsTerminal(int(os.Stdout.Fd())) {
```

Comment:
BTW we should also not ask for `globalIsJSON`  if set @aead

Reply:
👍